### DISCUSSION TITLE
xmake does not work with conda build tools

### DISCUSSION THREAD

**Discussion by jkawnkwh85** [xmake-io/xmake#3438] [General]
Having conda activated, xmake fails to link.

xmake version: xmake v2.7.6+master.c34ac1c69

The project is created simply using `xmake create test_project`

`xmake -rv` output
```
checking for /home/rnd/conda/bin/x86_64-conda-linux-gnu-c++ ... ok
checking for the c++ compiler (cxx) ... x86_64-conda-linux-gnu-c++
checking for /home/rnd/conda/bin/x86_64-conda-linux-gnu-c++ ... ok
checking for flags (-fPIC) ... no
checking for flags (-fvisibility-inlines-hidden) ... no
checking for flags (-O3) ... no
checking for flags (-DNDEBUG) ... no
[ 25%]: cache compiling.release src/main.cpp
/home/rnd/conda/bin/x86_64-conda-linux-gnu-c++ -c -m64 -fvisibility=hidden -o build/.objs/test_conda_boost/linux/x86_64/release/src/main.cpp.o src/main.cpp
checking for flags (-MMD -MF) ... no
checking for flags (-fdiagnostics-color=always) ... no
checking for flags (-fcolor-diagnostics) ... no
checking for /home/rnd/conda/bin/x86_64-conda-linux-gnu-ld ... ok
checking for the linker (ld) ... x86_64-conda-linux-gnu-ld
[ 50%]: linking.release test_conda_boost
/home/rnd/conda/bin/x86_64-conda-linux-gnu-ld -o build/linux/x86_64/release/test_conda_boost build/.objs/test_conda_boost/linux/x86_64/release/src/main.cpp.o -m64 -s
error: /home/rnd/conda/bin/x86_64-conda-linux-gnu-ld: unrecognised emulation mode: 64
Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu

warning: add_cxflags("-DNDEBUG") is ignored, please pass `{force = true}` or call `set_policy("check.auto_ignore_flags", false)` if you want to set it.
```

the `xmake.lua` file content

```
add_rules("mode.debug", "mode.release")

target("test_boost")
    set_kind("binary")
    set_toolset("ld", "g++")
    add_files("src/*.cpp")
```

the problem is resolved if I specify the linker explicitly.

```
add_rules("mode.debug", "mode.release")

target("test_conda_boost")
    set_kind("binary")
    set_toolset("ld", "g++")
    add_files("src/*.cpp")
```

It seems there is issue with detecting the correct linker with conda.

**Comment by waruqi** [comment]
The conda environment sets the $CC, $LD environment variable, which affects the choice of xmake's compilation toolchain, so xmake will use the LD=x86_64-conda-linux-gnu-ld specified by conda as the linker.

But normally, when compiling a c/c++ project, you should use clang++/g++ as the linker and do not use `ld` directly. And this toolchain seems to be a cross-compilation toolchain, not gcc/clang on the PC.

One solution is to unset the LD environment variable.

Another solution is to force xmake to switch to another toolchain, e.g. `xmake f --toolchain=gcc` or configure `set_toolchains("gcc")` in xmake.lua

**Reply by jkawnkwh85** [reply]
Thanks a lot for the clarification.

**Comment by waruqi** [comment]
try this patch https://github.com/xmake-io/xmake/pull/6022
